spec/components/game_board_component_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe GameBoardComponent, type: :component do
+  #pending "add some examples to (or delete) #{__FILE__}"
+  let(:game) {
+    Game.new(passcode: %w(red red green green).to_json)
+  }
+  let(:attempts) {
+    [
+      Attempt.new(values: %w(blue yellow orange purple)),
+      Attempt.new(values: %w(blue yellow orange red))
+    ]
+  }
+  let(:feedback) { GameBoardComponent.new(game: game, attempts: attempts )}
+
+  it "renders game board elements" do
+    render_inline feedback
+
+    aggregate_failures do
+      expect(page.find(".secret-row")).to have_text "????", exact: true
+      expect(page.all(".guess-chances").map(&:text)).to eq (1..10).to_a.map(&:to_s).reverse
+
+      expect(page.find(".color-picker").find(".bg-red")["data-color"]).to eq "red"
+      expect(page.find(".color-picker").find(".bg-orange")["data-color"]).to eq "orange"
+      expect(page.find(".color-picker").find(".bg-yellow")["data-color"]).to eq "yellow"
+      expect(page.find(".color-picker").find(".bg-green")["data-color"]).to eq "green"
+      expect(page.find(".color-picker").find(".bg-blue")["data-color"]).to eq "blue"
+      expect(page.find(".color-picker").find(".bg-purple")["data-color"]).to eq "purple"
+
+      expect(page).to have_text("Check")
+      expect(page).to have_text("✓")
+
+      bg_class = attempts.first.values
```

Comment:
Those value are `%w(blue yellow orange purple)`. That seems very different from a concept of `bg_class`.
Otherwise you're on the right track. Further down, checking for `data-number` values are not something relevant for rendering the elements correctly.
I assume what should be tested here, is that the first attempt is rendered correctly. To do that, it's worth checking that
the last `.guess-rating-box` contains four elements that have classes
`.guess-rating.bg-blue`, `.guess-rating.bg-yellow`, `.guess-rating.bg-orange`, `.guess-rating.bg-purple`.

The same thing goes for the second attempt.

For the general layout of the board, it would make sense to check that there are 10 "rows" of guesses, each containing 4 `.guess-rating` elements.

Reply:
As per this , ".guess-rating.bg-blue, .guess-rating.bg-yellow, .guess-rating.bg-orange, .guess-rating.bg-purple".

Do you mean the "inner-guess-cell" class instead of the guess-rating class? I am asking because the guess-rating class represents the feedback elements for each attempt of "no match, partial and exact matches." The "inner-guess-cell class represents the actual guesses.